### ISSUE TITLE
[Half-Life] Bullsquid crashes the game when NPC enemy dies at specific time

### ISSUE THREAD

**Issue by hobokenn** [ValveSoftware/halflife#2834]
Similar to https://github.com/ValveSoftware/halflife/issues/2088, but doesn't seem to work with the player himself, couldn't reproduce in OpFor or Blue Shift. [vid](https://youtu.be/HPtoNSQG3Q0) [save](https://github.com/ValveSoftware/halflife/files/3996485/quick.zip)

**Comment by SamVanheer** [comment]
It's probably this code:
https://github.com/ValveSoftware/halflife/blob/c7240b965743a53a29491dd49320c88eecf6257b/dlls/bullsquid.cpp#L531-L569

The Gonome code was based on this so it makes sense that it would have the same bugs.

**Comment by BlackShadow** [comment]
> It's probably this code:
> https://github.com/ValveSoftware/halflife/blob/c7240b965743a53a29491dd49320c88eecf6257b/dlls/bullsquid.cpp#L531-L569
> 
> The Gonome code was based on this so it makes sense that it would have the same bugs.

`vecSpitDir = ((m_hEnemy->pev->origin + m_hEnemy->pev->view_ofs) - vecSpitOffset).Normalize();`

This string causes the crash, i'm not sure why. Can't we create like a aimlock function for spitoffset to enemy's last position? Which will pretty much do the same thing.

**Comment by SamVanheer** [comment]
It's crashing because it's dereferencing the pointer to the enemy without checking if the pointer is non-null first. It's easy to fix, just use the same solution as the one used for the Gonome.

**Comment by hobokenn** [comment]
update: was able to reproduce the crash in op4 and blue shift, also i have noticed the same crash happens with pit drones in op4. i made a save a while ago thinking it was something else, if anyone wants to try it - [gotcha.zip](https://github.com/ValveSoftware/halflife/files/5292857/gotcha.zip). how it works: you load a save, make no inputs and if Otis is gibbed it will crash ~0.7s later, however if you move left fast enough pit drone in the back will notice you and the game won't crash.

**Comment by Solokiller** [comment]
Yep, Pit Drones have the exact same code, with the exact same bug.
